src/deepsparse/loggers/prometheus_logger.py

```diff
@@ -170,6 +199,33 @@ def get_prometheus_metric(
     )
 
 
+def _get_metric_from_the_mapping(
+    identifier: str, metric_type_mapping: Dict[str, str] = _IDENTIFIER_TO_METRIC_TYPE
+) -> Optional[Type["MetricWrapperBase"]]:  # noqa: F821
+    for system_group_name, metric_type in metric_type_mapping.items():
+        """
+        Attempts to get the metric type given the identifier and system_group_name.
+        There are two cases:
+        Case 1) If system_group_name contains both the group name and the identifier,
+            e.g. "request_details/successful_request", the match requires the identifier
+            to end with the system_group_name,
+            e.g. "pipeline_name/request_details/successful_request".
+        Case 2) If system_group_name contains only the group name,
+            e.g. "prediction_latency",
+            the match requires the system_group_name to be
+            contained within the identifier
+            e.g. prediction_latency/pipeline_inputs
+        """
+        if ("/" in system_group_name and identifier.endswith(system_group_name)) or (
+            system_group_name in identifier
+        ):
+            return metric_type
+    warnings.warn(
```

Comment:
I think that if we do that, the warning will "pop up" every time we are encountering a log that cannot be processed by Prometheus. I chose to warn here, so that the warning happens once at the attempt to "register" the mapping from identifier to Prometheus metric, and never again (not to clutter the monitoring service)